go-controller/pkg/ovn/gateway.go

```diff
@@ -425,7 +425,7 @@ func (gw *GatewayManager) GatewayInit(
 			types.TopologyExternalID: gw.netInfo.TopologyType(),
 		}
 		_, isNetIPv6 := gw.netInfo.IPMode()
-		if gw.netInfo.IsPrimaryNetwork() && gw.netInfo.TopologyType() == types.Layer2Topology && isNetIPv6 && config.IPv6Mode {
+		if gw.netInfo.TopologyType() == types.Layer2Topology && isNetIPv6 && config.IPv6Mode {
```

Comment:
Would you elaborate on why this is not necessary ? (in the commit msg)

I understand that we only invoke GW init for primary UDNs IIF the feat gate is enabled FWIW.

But having it in the commit msg + a link to the callers could be invaluable in a few months / years.

Reply:
yup will add this to the commit message to make it clear.

Reply:
done!